src/mongo_odm/test/query_builder.cpp

```diff
@@ -961,6 +1021,19 @@ TEST_CASE("Update Builder", "mongo_odm::update_expr") {
 
     /* Array update operators */
 
+    SECTION("Test $ array update operator.", "[mongo_odm::dollar_operator_nvp]") {
+        auto res =
+            coll.update_one((MONGO_ODM_KEY(Bar::w) == 555,
+                             MONGO_ODM_KEY(Bar::arr).elem_match(MONGO_ODM_ELEM(Bar::arr) == 5)),
```

Comment:
Currently, without `elem_match`, the query becomes `{w: {$eq: 555}, $eq: 5}`, which is syntactically correct. 
One way to solve this would be to allow something like `MONGO_ODM_KEY(Bar::arr) == 5`, which would be comes `{arr: {$eq: 5}}`. Right now this is disabled because array fields must be compared to array values, but I can put it a ticket to allow arrays to be directly compared to a value which is the type of their element. So `Bar::arr`, a vector of `int`s, could be compared to an `int` value like `5`. Which is syntactically valid for Mongo queries, I just didn't realize that was possible at the time.